src/constraint/spec.ts

```diff
@@ -232,7 +232,9 @@ export const SPEC_CONSTRAINTS: SpecConstraintModel[] = [
         // channel unspecified, no need to check
         if (isWildcard(encQ.channel)) return true;
 
-        return !!supportMark(encQ.channel, mark as Mark);
+        if (encQ.channel === 'row' || encQ.channel === 'column') return true;
+
+        return !!supportMark(encQ.channel as CHANNEL.Channel, mark as Mark);
```

Comment:
Why do we need to force CHANNEL.Channel? If you extract a channel variable, typescript should be able to figure out that channel isn't row or column anymore.

Reply:
I get a build error saying essentially that ExtendedChannel !== Channel when I do not cast the type of encQ.channel to CHANNEL.Channel. I changed encQ.channel to be of type ExtendedChannel to accommodate faceting.

Reply:
Then supportMark should support extended channel as well. If we just cast here, we may be passing an extended channel to something that supports only channel, which will cause bugs.